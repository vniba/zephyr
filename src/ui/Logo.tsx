import styled from 'styled-components';

const StyledLogo = styled.div`
  margin: 0 auto;
`;

const Img = styled.img`
  height: 9.6rem;
  width: auto;
  border-radius: 50%;
  @media (max-width: 800px) {
    height: 4rem;
  }
`;

function Logo() {
  return (
    <StyledLogo>
      <Img src='/img/logo.svg' alt='Logo' />
    </StyledLogo>
  );
}

export default Logo;
