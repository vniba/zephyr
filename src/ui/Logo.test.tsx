import { render, screen } from '@testing-library/react';
import Logo from './Logo.tsx';
import { expect } from 'vitest';

describe('logo component', function () {
  it('should render img tag with logo', function () {
    render(<Logo />);
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.getByAltText(/logo/i)).toBeInTheDocument();
  });

  it('should show correct styles', function () {
    const { container } = render(<Logo />);
    expect(container.firstChild).toHaveStyleRule('height:9.6rem');
  });

  it('should  match default Logo snapshot ', function () {
    const logo = render(<Logo />);
    expect(logo).toMatchSnapshot();
  });
});
