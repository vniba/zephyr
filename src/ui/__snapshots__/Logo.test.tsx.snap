// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`logo component > should  match default Logo snapshot  1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  text-align: center;
}

.c1 {
  height: 9.6rem;
  width: auto;
  border-radius: 50%;
}

<body>
    <div>
      <div
        class="c0"
      >
        <img
          alt="Logo"
          class="c1"
          src="/img/logo.svg"
        />
      </div>
    </div>
  </body>,
  "container": .c0 {
  text-align: center;
}

.c1 {
  height: 9.6rem;
  width: auto;
  border-radius: 50%;
}

<div>
    <div
      class="c0"
    >
      <img
        alt="Logo"
        class="c1"
        src="/img/logo.svg"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
